helpers_test.go

```diff
@@ -35,3 +38,149 @@ func requireTestmod(tb testing.TB) {
 		tb.Skip("bpf_testmod not loaded")
 	}
 }
+
+func newMap(tb testing.TB, spec *MapSpec, opts *MapOptions) (*Map, error) {
+	tb.Helper()
+
+	if opts == nil {
+		opts = new(MapOptions)
+	}
+
+	m, err := NewMapWithOptions(spec, *opts)
+	testutils.SkipIfNotSupportedOnOS(tb, err)
+	if err != nil {
+		return nil, err
+	}
+
+	tb.Cleanup(func() { m.Close() })
+	return m, nil
+}
+
+func mustNewMap(tb testing.TB, spec *MapSpec, opts *MapOptions) *Map {
+	tb.Helper()
+
+	m, err := newMap(tb, spec, opts)
+	qt.Assert(tb, qt.IsNil(err))
+
+	return m
+}
+
+func createMap(tb testing.TB, typ MapType, maxEntries uint32) *Map {
+	tb.Helper()
+
+	return mustNewMap(tb, &MapSpec{
+		Name:       "test",
+		Type:       typ,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: maxEntries,
+	}, nil)
+}
+
+func createMapInMap(tb testing.TB, outer, inner MapType) *Map {
+	tb.Helper()
+
+	return mustNewMap(tb, &MapSpec{
+		Type:       outer,
+		KeySize:    4,
+		MaxEntries: 2,
+		InnerMap: &MapSpec{
+			Type:       inner,
+			KeySize:    4,
+			ValueSize:  4,
+			MaxEntries: 2,
+		},
+	}, nil)
+}
+
+func newProgram(tb testing.TB, spec *ProgramSpec, opts *ProgramOptions) (*Program, error) {
+	tb.Helper()
+
+	if opts == nil {
+		opts = new(ProgramOptions)
+	}
+
+	if spec.Type == SkLookup && spec.AttachType == AttachNone {
```

Comment:
Assuming this is for `TestProgramRunEmptyData`, I'd make that test call `newProgram` instead so it can provide a spec with the right attach type. This seems a little heavy-handed.